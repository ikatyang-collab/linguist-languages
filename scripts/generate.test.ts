import { run } from './generate'
import { wrap } from 'jest-snapshot-serializer-raw'

const fakeLanguagesYml = `
# Defines all Languages known to GitHub.
#
# fs_name               - Optional field. Only necessary as a replacement for the sample directory name if the
#                         language name is not a valid filename under the Windows filesystem (e.g., if it
#                         contains an asterisk).
# type                  - Either data, programming, markup, prose, or nil
# aliases               - An Array of additional aliases (implicitly
#                         includes name.downcase)
# ace_mode              - A String name of the Ace Mode used for highlighting whenever
#                         a file is edited. This must match one of the filenames in http://git.io/3XO_Cg.
#                         Use "text" if a mode does not exist.
#
# Any additions or modifications (even trivial) should have corresponding
# test changes in \`test/test_blob.rb\`.
#
# Please keep this list alphabetized. Capitalization comes before lowercase.

Test1:
  type: programming
  color: "#814CCC"
  aliases:
  - hello
  extensions:
  - ".bsl"
  - ".os"
  tm_scope: source.bsl
  ace_mode: text
  language_id: 0
Test Test 2:
  type: programming
  color: "#E8274B"
  extensions:
  - ".abap"
  ace_mode: abap
  wrap: true
  language_id: 1
F*:
  fs_name: Fstar
  type: programming
  color: "#572e30"
  aliases:
  - fstar
  extensions:
  - ".fst"
  tm_scope: source.fstar
  ace_mode: text
  language_id: 336943375
`

const writes: [string, string][] = []

test('run', () => {
  run({
    clean: false,
    read: () => fakeLanguagesYml,
    write: (filename, content) => writes.push([filename, content]),
  })

  writes.forEach(([filename, content]) => {
    expect(wrap(content)).toMatchSnapshot(
      filename.replace(process.cwd(), '<cwd>'),
    )
  })
})
