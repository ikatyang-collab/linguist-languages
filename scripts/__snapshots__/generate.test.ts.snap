// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`run > <cwd>/data/Fstar.js 1`] = `
export default {
  "name": "F*",
  "fsName": "Fstar",
  "type": "programming",
  "color": "#572e30",
  "aliases": [
    "fstar"
  ],
  "extensions": [
    ".fst"
  ],
  "tmScope": "source.fstar",
  "aceMode": "text",
  "languageId": 336943375
}
`;

exports[`run > <cwd>/data/Test Test 2.js 1`] = `
export default {
  "name": "Test Test 2",
  "type": "programming",
  "color": "#E8274B",
  "extensions": [
    ".abap"
  ],
  "aceMode": "abap",
  "wrap": true,
  "languageId": 1
}
`;

exports[`run > <cwd>/data/Test1.js 1`] = `
export default {
  "name": "Test1",
  "type": "programming",
  "color": "#814CCC",
  "aliases": [
    "hello"
  ],
  "extensions": [
    ".bsl",
    ".os"
  ],
  "tmScope": "source.bsl",
  "aceMode": "text",
  "languageId": 0
}
`;

exports[`run > <cwd>/lib/index.d.ts 1`] = `
type LanguageName =
  | "Test1"
  | "Test Test 2"
  | "F*"

export interface Language {
  name: string
  /**
   * Either data, programming, markup, prose, or nil
   */
  type: string
  color: string
  /**
   * An Array of additional aliases (implicitly
   * includes name.downcase)
   */
  aliases?: string[]
  extensions: string[]
  tmScope?: string
  /**
   * A String name of the Ace Mode used for highlighting whenever
   * a file is edited. This must match one of the filenames in http://git.io/3XO_Cg.
   * Use "text" if a mode does not exist.
   */
  aceMode: string
  languageId: number
  wrap?: boolean
  /**
   * Optional field. Only necessary as a replacement for the sample directory name if the
   * language name is not a valid filename under the Windows filesystem (e.g., if it
   * contains an asterisk).
   */
  fsName?: string
}

declare const languages: Record<LanguageName, Language>

export default languages
`;

exports[`run > <cwd>/lib/index.js 1`] = `
import _0 from "../data/Test1.js"
import _1 from "../data/Test%20Test%202.js"
import _2 from "../data/Fstar.js"
export default {
  "Test1": _0,
  "Test Test 2": _1,
  "F*": _2,
}
`;
